protocol/messenger.go

```diff
@@ -4692,54 +4724,88 @@ func (m *Messenger) MarkAllReadInCommunity(communityID string) ([]string, error)
 
 // MuteChat signals to the messenger that we don't want to be notified
 // on new messages from this chat
-func (m *Messenger) MuteChat(chatID string) error {
-	chat, ok := m.allChats.Load(chatID)
+func (m *Messenger) MuteChat(request *requests.MuteChat) (time.Time, error) {
+	chat, ok := m.allChats.Load(request.ChatID)
 	if !ok {
 		// Only one to one chan be muted when it's not in the database
-		publicKey, err := common.HexToPubkey(chatID)
+		publicKey, err := common.HexToPubkey(request.ChatID)
 		if err != nil {
-			return err
+			return time.Time{}, err
 		}
 
 		// Create a one to one chat and set active to false
-		chat = CreateOneToOneChat(chatID, publicKey, m.getTimesource())
+		chat = CreateOneToOneChat(request.ChatID, publicKey, m.getTimesource())
 		chat.Active = false
 		err = m.initChatSyncFields(chat)
 		if err != nil {
-			return err
+			return time.Time{}, err
 		}
 		err = m.saveChat(chat)
 		if err != nil {
-			return err
+			return time.Time{}, err
 		}
 	}
 
 	var contact *Contact
 	if chat.OneToOne() {
-		contact, _ = m.allContacts.Load(chatID)
+		contact, _ = m.allContacts.Load(request.ChatID)
+	}
+
+	MuteTill := time.Time{}
+
+	switch request.MutedType {
+	case MuteTill1Min:
+		MuteTill = time.Now().Add(time.Duration(MuteFor1MinDuration))
```

Comment:
@ibrkhalil, why do we need to cast to a duration if the constant is already a `time.Duration`?

Reply:
I can't remember exactly why I did it, let me test and reply

Reply:
Removed

Reply:
Turned out to be unnecessary .. Removed